Fix compile errors for gcc-14 on Debian trixie

lvk88 opened the pull request:
Currently it is not possible to build lib3mf master on Debian Trixie with gcc 14. This is because  inclusion of `<algorithm>` is needed in order to use std::find, but it was missing, leading to the compiler error.

This pull request fixes this issue.

vijaiaeroastro commented:
@lvk88 https://github.com/3MFConsortium/lib3mf/pull/413 already fixes the issue in develop.

lvk88 committed:
Fix compile errors for gcc-14

Inclusion of <algorithm> is needed in order to use std::find, but it was
missing, leading to compiler errors with gcc-14 on Debian Trixie.

lvk88 commented:
Hey @vijaiaeroastro , thanks for pointing this out, I wasn't aware. #413 only partially fixes the compiler error, `#include <algorithm>` is missing from another file as well. I opened #421 to fix that error.
